zebra-consensus/src/block/subsidy/general.rs

```diff
@@ -69,41 +68,7 @@ pub fn block_subsidy(height: Height, network: Network) -> Result<Amount<NonNegat
     }
 }
 
-/// `MinerSubsidy(height)` as described in [protocol specification §7.8][7.8]
-///
-/// [7.8]: https://zips.z.cash/protocol/protocol.pdf#subsidies
-///
-/// `non_miner_reward` is the founders reward or funding stream value.
-/// If all the rewards for a block go to the miner, use `None`.
-#[allow(dead_code)]
-pub fn miner_subsidy(
-    height: Height,
-    network: Network,
-    non_miner_reward: Option<Amount<NonNegative>>,
-) -> Result<Amount<NonNegative>, Error> {
-    if let Some(non_miner_reward) = non_miner_reward {
-        block_subsidy(height, network)? - non_miner_reward
-    } else {
-        block_subsidy(height, network)
-    }
-}
-
-/// Returns a list of outputs in `Transaction`, which have a value equal to `Amount`.
-pub fn find_output_with_amount(
```

Comment:
Is this not used for funding streams, or was this duplicative of that code?

Reply:
The funding streams are doing it a bit different than the code we had for founders rewards. We could unify some code but as we removed founders reward that function is not needed anymore.

The funding stream validation logic starts here: https://github.com/ZcashFoundation/zebra/blob/19bfa26c80d27777fca710f0e319665184e15367/zebra-consensus/src/block/check.rs#L132 where the amounts and receivers(currently ecc, fundation, major grants) for a given height are calculated from the protocol equations.
Then for each receiver and value the address is obtained, again from protocol formulas (https://github.com/ZcashFoundation/zebra/blob/19bfa26c80d27777fca710f0e319665184e15367/zebra-consensus/src/block/check.rs#L145).
No check in the transaction itself has been done yet, once we have those 2 things calculated then the expected value and expected addresses are searched in the coinbase transaction outputs (https://github.com/ZcashFoundation/zebra/blob/19bfa26c80d27777fca710f0e319665184e15367/zebra-consensus/src/block/check.rs#L148-L151). If both are there and in the same output then the funding stream is valid.